src/bft.py

```diff
@@ -1,15 +1,25 @@
-"""A module for breadth-first traversal of trees."""
-
-from collections import deque
-from typing import Iterable
-from tree import T
-
-
-def bf_order(t: T | None) -> Iterable[int]:
-    """Breadth-first traversal of a tree.
-
-    >>> tree = T(2, T(1, None, None), T(4, T(3, None, None), T(5, None, None)))
-    >>> list(bf_order(tree))
-    [2, 1, 4, 3, 5]
-    """
-    return  # FIXME
+"""A module for breadth-first traversal of trees."""
+
+from collections import deque
+from typing import Iterable
+from tree import T
+
+
+def bf_order(t: T or None) -> Iterable[int]:
+    if t == None:
+        return []
+
+    node_queue = [t]

```

Comment:
Use a `deque` here unless you want quadratic running time. When you `pop(0)` later, you get an expensive operation. If you use a proper queue, you won't.